### 2024年12月28日
## mashuai

在写dwd的时候运行代码的时候会报一个错误:
onnector'='hbase-2. table-name'='amall:dim base dic'='cdhe1,cdhe2,cdne3 F1n女中的1由卡为在中病:util.iava:167)1T口。 actoryUtil.iawa:192)ceTable.java:175)
nicTablesource(catalogSc Rel(catalogsourceTable.iava:115)8:3743)Converter.java:2666)r.java:2233)java:2092]va:2745)AVA:2708 ava:2147:2092 1:3025 ava:2252 java:2092
:700).iava:686)rter.java:3589)1:599]at org.apache ble$planner$calcite$flinkPlannerImpl$$rel(FlinkPlannerImpl.scala:216)
at org scala:192)at or n(sqlTo0peratior verter.java:1580)at g vertsalQuery(salTo0perationconverter.iava:1285)
ertValidatedsqlNode(sqlTo0perationConverter.java:397)vert(SqlTooperationConverter.java:282)at ol
at org.… .java:106)at org.apache.flin!onmentImpl.iava:738)at retailersv1. tInfo.java:43)at retailersv1.
Info,iava:35)Cannot discover a connector using option: "connector'='hbase-2.4' used by: org.apache at org.apache.flink.
nectorError(Factoryutil.java:736)at org.apache,flink torwutil.iava:710)at org.apache.flink(Factoryutil.iava:163)... 28 more
aused by: org.apache.flink.table.api Could not find any factory for identifier 'hbase-2.4' that implements'org.apache,flink,table.factories.DynamicTableFactory' in the classps

在类中找不到标识符“hbease2.4”现在还没有解决,但是换了一种方法去写;
期间报了一个找不到ts,仔细观察数据后发现数据中的没有ts只能找到ts_ms,ts改为ts_ms即可;

正在报一个这个错,正在解决中:
Caused by: java.util.concurrent.ExecutionException: org.apache.kafka.common.errors.TimeoutException: Timed out waiting for a node assignment. Call: describeTopics
at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)
at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1908)
at org.apache.kafka.common.internals.KafkaFutureImpl.get(KafkaFutureImpl.java:165)
at org.apache.flink.connector.kafka.source.enumerator.subscriber.KafkaSubscriberUtils.getTopicMetadata(KafkaSubscriberUtils.java:44)
... 10 more
Caused by: org.apache.kafka.common.errors.TimeoutException: Timed out waiting for a node assignment. Call: describeTopics
